File: core/src/dc/targetman/mechanics/Weapon.java
package dc.targetman.mechanics;

import dclib.util.FloatRange;

public final class Weapon {

	private final float reloadTime;
	private final int numBullets;
	private final float spread;
	private final FloatRange speedRange;
	private final String bulletType;

	public Weapon(final float reloadTime, final int numBullets, final float spread, final float minSpeed, 
			final float maxSpeed, final String bulletType) {
		this.reloadTime = reloadTime;
		this.numBullets = numBullets;
		this.spread = spread;
		this.speedRange = new FloatRange(minSpeed, maxSpeed);
		this.bulletType = bulletType;
	}

	public final float getReloadTime() {
		return reloadTime;
	}

	public final int getNumBullets() {
		return numBullets;
	}

	public final float getSpread() {
		return spread;
	}
	
	public final FloatRange getSpeedRange() {
		return speedRange;
	}

	public final String getBulletType() {
		return bulletType;
	}

}

File: core/src/main/dc/targetman/mechanics/weapon/Weapon.kt
package dc.targetman.mechanics.weapon

import com.badlogic.gdx.math.Vector2
import com.esotericsoftware.spine.Skeleton
import dc.targetman.skeleton.bounds
import dclib.geometry.size
import dclib.util.FloatRange
import dclib.util.Timer

class Weapon(val data: WeaponData, val skeleton: Skeleton) {
    val reloadTimer = Timer(data.reloadTime, data.reloadTime)
    val speedRange = FloatRange(data.minSpeed, data.maxSpeed)

    val size: Vector2
        get() {
            val size = skeleton.bounds.size
            val weaponHeight = data.width * size.y / size.x
            return Vector2(data.width, weaponHeight)
        }
}